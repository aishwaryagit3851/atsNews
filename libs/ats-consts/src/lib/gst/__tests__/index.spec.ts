import {
  GST_SLABS_OBJ,
  TGST_SLABS,
  GST_SLABS,
  GST_PERCENTAGES,
  TaxApplicableTypes,
  TTaxApplicableTypes,
  GST_STATE_OF_SUPPLY_OBJ,
  GST_STATE_OF_SUPPLY
} from '..';

describe('ATS CONSTS for - gst', () => {
  it('Match consts to snapshot', () => {
    expect(GST_SLABS_OBJ).toMatchSnapshot();

    const NewTGST_SLABS: TGST_SLABS[] = [
      '0',
      '12',
      '18',
      '28',
      '5',
      'EXEMPTED',
      'NILL'
    ];
    expect(NewTGST_SLABS).toMatchSnapshot();

    expect(GST_SLABS).toMatchSnapshot();
    expect(GST_PERCENTAGES).toMatchSnapshot();
    expect(TaxApplicableTypes).toMatchSnapshot();

    const NewTTaxApplicableTypes: TTaxApplicableTypes[] = [
      'EXCLUSIVE',
      'INCLUSIVE'
    ];
    expect(NewTTaxApplicableTypes).toMatchSnapshot();

    expect(GST_STATE_OF_SUPPLY_OBJ).toMatchSnapshot();
    expect(GST_STATE_OF_SUPPLY).toMatchSnapshot();
  });
});
