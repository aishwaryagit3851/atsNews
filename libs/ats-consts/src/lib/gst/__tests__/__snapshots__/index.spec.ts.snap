// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ATS CONSTS for - gst Match consts to snapshot 1`] = `
{
  "0": "0",
  "0.25": "0.25",
  "12": "12",
  "18": "18",
  "28": "28",
  "3": "3",
  "5": "5",
  "EXEMPTED": "EXEMPTED",
  "NILL": "NILL",
}
`;

exports[`ATS CONSTS for - gst Match consts to snapshot 2`] = `
[
  "0",
  "12",
  "18",
  "28",
  "5",
  "EXEMPTED",
  "NILL",
]
`;

exports[`ATS CONSTS for - gst Match consts to snapshot 3`] = `
[
  "0",
  "3",
  "5",
  "12",
  "18",
  "28",
  "EXEMPTED",
  "NILL",
  "0.25",
]
`;

exports[`ATS CONSTS for - gst Match consts to snapshot 4`] = `
[
  {
    "label": "EXEMPTED",
    "value": "EXEMPTED",
  },
  {
    "label": "NILL",
    "value": "NILL",
  },
  {
    "label": "0%",
    "value": "0",
  },
  {
    "label": "0.25%",
    "value": "0.25",
  },
  {
    "label": "3%",
    "value": "3",
  },
  {
    "label": "5%",
    "value": "5",
  },
  {
    "label": "12%",
    "value": "12",
  },
  {
    "label": "18%",
    "value": "18",
  },
  {
    "label": "28%",
    "value": "28",
  },
]
`;

exports[`ATS CONSTS for - gst Match consts to snapshot 5`] = `
[
  "INCLUSIVE",
  "EXCLUSIVE",
]
`;

exports[`ATS CONSTS for - gst Match consts to snapshot 6`] = `
[
  "EXCLUSIVE",
  "INCLUSIVE",
]
`;

exports[`ATS CONSTS for - gst Match consts to snapshot 7`] = `
{
  "INTER": "INTER",
  "INTRA": "INTRA",
}
`;

exports[`ATS CONSTS for - gst Match consts to snapshot 8`] = `
[
  "INTRA",
  "INTER",
]
`;
